### PR TITLE
Add an env var to disable pkg_resources

### PR DESCRIPTION
**Pull request by ssteinbach**
- in some cases, pkg_resources can be slow to both import and use
- adds the `OTIO_DISABLE_PKG_RESOURCE_PLUGINS` environment variable to
  disable using this library for users that have those issues.
- Even for users not experiencing those issues, this speeds up the
  import time of OTIO slightly.
- Also adds a new doc page with a description of OTIO environment variables.

### CODE REVIEW

File: docs/tutorials/otio-env-variables.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Environment Variables
+
+This document describes the environment variables that can be used to configure
+various aspects of OTIO.
+
+## Plugin Configuration
+
+These variables must be set _before_ the OpenTimelineIO python library is imported.
+
+- `OTIO_PLUGIN_MANIFEST_PAT`H: a ":" separated string with paths to .manifest.json files that contain OTIO plugin manifests.  See: <a href="write-an-adapter.html" target="_blank">Tutorial on how to write an adapter plugin</a>.
```
Comment (meshula) [inline docs/tutorials/otio-env-variables.md:10]: typo: _PAT`H

File: docs/tutorials/otio-env-variables.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- `OTIO_DEFAULT_MEDIA_LINKER`: the name of the default media linker to use after reading a file, if "" then no media linker is automatically invoked.
+- O`TIO_DISABLE_PKG_RESOURCE_PLUGINS`: By default, OTIO will use the pkg_resource entry_points mechanism to discover plugins that have been installed into the current python environment.  pkg_resources, however, can be slow in certain cases, so for users who wish to disable this behavior, this variable can be set to 1.
```
Comment (meshula) [inline docs/tutorials/otio-env-variables.md:12]: typo O`TIO

File: src/py-opentimelineio/opentimelineio/plugins/manifest.py [hunk 3]
```diff
@@ -28,11 +28,17 @@
 import logging
 import os
 
-# on some python interpreters, pkg_resources is not available
-try:
-    import pkg_resources
-except ImportError:
+# pkg_resources can have bad performance characteristics in some circumstances
```
Comment (meshula) [inline src/py-opentimelineio/opentimelineio/plugins/manifest.py:31]: clarity/punctuation:

In some circumstances pkg_resources has bad performance characteristics. Using this flag: OTIO_DISABLE_PKG_RESOURCE_PLUGINS disables OpenTimelineIO's import and of use of the pkg_resources system.

**Review by meshula — CHANGES_REQUESTED** [review]
minor typos

File: docs/tutorials/otio-env-variables.md (hunk 1 above)
Comment (ssteinbach) [reply docs/tutorials/otio-env-variables.md:10]: Cool.  Using the "insert a suggestion" feature to correct these typos.

```suggestion
- `OTIO_PLUGIN_MANIFEST_PATH`: a ":" separated string with paths to .manifest.json files that contain OTIO plugin manifests.  See: <a href="write-an-adapter.html" target="_blank">Tutorial on how to write an adapter plugin</a>.
```

File: docs/tutorials/otio-env-variables.md (hunk 2 above)
Comment (ssteinbach) [reply docs/tutorials/otio-env-variables.md:12]:
```suggestion
- `OTIO_DISABLE_PKG_RESOURCE_PLUGINS`: By default, OTIO will use the pkg_resource entry_points mechanism to discover plugins that have been installed into the current python environment.  pkg_resources, however, can be slow in certain cases, so for users who wish to disable this behavior, this variable can be set to 1.
```

**Review by meshula — APPROVED** [review]

**Pull request merged**